xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-test/xwiki-platform-appwithinminutes-test-pageobjects/src/main/java/org/xwiki/appwithinminutes/test/po/ApplicationHomePage.java

```diff
@@ -73,7 +73,9 @@ public static ApplicationHomePage gotoPage(String appName)
      */
     public EntryNamePane clickAddNewEntry()
     {
-        addEntryLink.click();
+        this.addEntryLink.click();
+        // Waits for the dialog box to be displayed before continuing.
+        getDriver().waitUntilElementIsVisible(By.cssSelector("div.xdialog-box"));
```

Comment:
This waits is not provided by `EntryNamePane` constructor?

Reply:
Not currently, but we could introduce a constructor on EntryNamePane with this wait if it's a better place for it.

Reply:
I'm fine with having it here, it might make it more difficult to use when adding the wait directly in constructor.